zulipterminal/ui_tools/buttons.py

```diff
@@ -647,6 +647,40 @@ def handle_narrow_link(self) -> None:
                 self.controller.exit_popup()
 
 
+class CodeSnippetButton(urwid.Text):
```

Comment:
As mentioned elsewhere, this works, but it would be good to migrate to a button with the specific functionality we want.

Reply:
I did trying inherting urwid.Button and Moving the behavior into the CodeSnippetButton but I could only move the method get_code_from_snippet() from views.py to buttons.py, since all other methods require the instance of controller to be taken into account. Kindly let me know anyother way to do this.

Reply:
Right now, all the basic Button functionality currently in `CodeSnippetButton` except for `get_code_from_snippet` is as per the default in `urwid.Button` (you can compare yourself, to `wimp.py` in urwid).

Look at the initializer for `MessageLinkButton` - that is how we can set up a controller via the button. We may also want to store the code content (snippet information) in there, for example.

Right now you're using the class name to call a method in a class, and explicitly passing the `self` value. Normally that'd be a bad idea and often cause an error, but you're not using `self` in `get_code_from_snippet` here! Instead, we should be creating `CodeSnippetButton`s with the relevant data, then using their properties as we need them.

I understand you've inherited this PR, but this (and the UI, using urwid) is using OOP in Python, so you may need to learn more about this if you've not worked on this before.